packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+/**
+ * React hook that measures virtualized space based on a static size to ensure optimized virtualization length.
+ */
+export const useStaticVirtualizerMeasure = (
+  defaultItemSize: number,
+  scrollView: HTMLElement | null = null,
+  direction: 'vertical' | 'horizontal' = 'vertical',
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+} => {
+  const [virtualizerLength, setVirtualizerLength] = React.useState(0);
+  const [virtualizerBufferItems, setVirtualizerBufferItems] = React.useState(0);
+  const [virtualizerBufferSize, setVirtualizerBufferSize] = React.useState(0);
+
+  const { targetDocument } = useFluent();
+
+  // If no ref is provided, the virtualizer should be in the main body document's scroll.
+  const _scrollView = scrollView ?? targetDocument?.documentElement ?? null;
```

Comment:
this hook should return a `ref` the examples currently work because there are multiple renders of the entire example component so the ref is set correctly on the second render. In cases where there is only one initial render, the scrollview will never be set and the default document will be used

Reply:
We also tend to adopt solutions that avoid this kind of double re-rending to initialize DOM elements and such since the perf hit of that can bloat apps